exporter/sapmexporter/correlation.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sapmexporter
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/signalfx/signalfx-agent/pkg/apm/correlations"
+	"github.com/signalfx/signalfx-agent/pkg/apm/tracetracker"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+// Tracker correlation
+type Tracker struct {
+	once         sync.Once
+	log          *zap.Logger
+	cfg          *Config
+	params       component.ExporterCreateParams
+	traceTracker *tracetracker.ActiveServiceTracker
+	correlation  *correlationContext
+}
+
+type correlationContext struct {
+	correlations.CorrelationClient
+	cancel context.CancelFunc
+}
+
+// NewTracker creates a new tracker instance for correlation.
+func NewTracker(cfg *Config, params component.ExporterCreateParams) *Tracker {
+	return &Tracker{
+		log:    params.Logger,
+		cfg:    cfg,
+		params: params,
+	}
+}
+
+func newCorrelationClient(cfg *Config, params component.ExporterCreateParams) (
+	*correlationContext, error,
+) {
+	endpoint := cfg.APIEndpoint
+	if cfg.Correlation.Endpoint != "" {
+		endpoint = cfg.Correlation.Endpoint
+	}
+
+	corrURL, err := url.Parse(endpoint)
```

Comment:
Does something validate that endpoint is not blank?  `urlParse("") == url.URL{}`